### PR TITLE
Adding Math Logic to Calculator Component

### PR DESCRIPTION
**Pull request by MarcoDDM**
I have implemented the math logic for the Calculator component, which was provided in the files calculate.js and operate.js. Here are the steps I took to complete this task:

**Item List:**

- Created a new directory called logic inside the project.
- Copied the files calculate.js and operate.js into the logic/ directory.
- Installed big.js using NPM.
- Analyzed the files calculate.js and operate.js to understand their functionality.
- Imported the files into the Calculator.js component.
- Implemented click event handlers using the math logic from calculate.js and operate.js in the React component.
- Tested the app in the browser and ensured that the Calculator is now able to perform math operations.

The updated code has been pushed to a new branch, and I have thoroughly tested it to make sure it works as expected. Please review and let me know your feedback. Thanks for your review and let me know if there any changes needed.

### CODE REVIEW

File: src/components/Calculator.js [hunk 1]
```diff
@@ -1,39 +1,60 @@
...
+        <button className="calculatorButton mathButton" type="button" onClick={() => handleButtonClick('3')}>3</button>
+        <button className="mathButton calculatorButton operationButton" type="button" onClick={() => handleButtonClick('+')}>
+          +
+        </button>
+        <button className="calculatorButton zero mathButton" type="button" onClick={() => handleButtonClick('0')}>0</button>
+        <button className="calculatorButton mathButton" type="button" onClick={() => handleButtonClick('.')}>.</button>
+        <button className="mathButton calculatorButton equalsButton" type="button" onClick={() => handleButtonClick('=')}>
+          =
+        </button>
       </div>
```
Comment (julie-ify) [inline src/components/Calculator.js]: - [ ] You were required to create a child component in a previous project. Now in this project, you are required to pass props to the child component. I cannot find any child components here. Kindly create a component and use it as a child of this calculator component in any way you deem fit, but the state should remain in this Calculator component.

**Review by julie-ify — CHANGES_REQUESTED** [review]
## Hi @MarcoDDM,

Good job so far!

## Highlights!
- Good feature branch name ✔️ 
- No linter errors ✔️ 
- Descriptive pull request ✔️ 
- Descriptive commit messages ✔️ 
- Good code organization and working app ✔️
- Descriptive README ✔️ 

Hello ✋, you have done well reaching this point of the milestone 🤝 . But there are some issues that you still need to work on to go to the next project but you are almost there!

# Status: Required Changes ♻️⛔️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread and tag me `@julie-ify` if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/components/Calculator.js [hunk 2]
```diff
@@ -2,7 +2,13 @@ import React, { useState } from 'react';
 import './Calculator.css';
 import calculate from '../logic/calculate';
 
-function Calculator() {
+const Calculator = () => (
+  <div className="container">
+    <ChildComponent />
+  </div>
+);
```
Comment (ridwanediallo) [inline src/components/Calculator.js]: - [ ] Excellent attempt to implement the previous reviewer's requested change :clap: . However, you are not passing any props to the child component in this manner. To complete this task, I recommend saving the child component in a separate file and importing it into the Calculator component, where you can pass the start through props from the Calculator component to the child component. The calculator component should include the state. Here are some suggestions to assist you. :point_down: 


### Follow these steps: :point_down: 

1. First create a separate file called `ChildComponent.js` and put this :point_down:  there
#### Note: 
Notice we are using `PropTypes` to fix the props validation so you need to run this command `npm i -S prop-types`  to install `prop-types`
```
import React from 'react';
import PropTypes from 'prop-types';

function ChildComponent({ calcData, handleButtonClick }) {
  return (
    <div className="calculator container">
      <input
        className="inputContainer"
        type="text"
        id="myInput"
        name="name"
        pattern="[0-9+\-*/(). ]+"
        value={calcData.next || calcData.total || ''}
        onChange={() => {}}
      />
      <div className="buttonContainer">
        <button
          className="calculatorButton mathButton"
          type="button"
          onClick={() => handleButtonClick('AC')}
        >
          AC
        </button>
        <button
          className="calculatorButton mathButton"
          type="button"
          onClick={() => handleButtonClick('+/-')}
        >
          +/-
        </button>
        <button
          className="calculatorButton mathButton"
          type="button"
          onClick={() => handleButtonClick('%')}
        >
          %
        </button>
        <button
          className="mathButton calculatorButton operationButton"
          type="button"
          onClick={() => handleButtonClick('÷')}
        >
          ÷
        </button>
        <button
          className="calculatorButton mathButton"
          type="button"
          onClick={() => handleButtonClick('7')}
        >
          7
        </button>
        <button
          className="calculatorButton mathButton"
          type="button"
          onClick={() => handleButtonClick('8')}
        >
          8
        </button>
        <button
          className="calculatorButton mathButton"
          type="button"
          onClick={() => handleButtonClick('9')}
        >
          9
        </button>
        <button
          className="mathButton calculatorButton operationButton"
          type="button"
          onClick={() => handleButtonClick('x')}
        >
          x
        </button>
        <button
          className="calculatorButton mathButton"
          type="button"
          onClick={() => handleButtonClick('4')}
        >
          4
        </button>
        <button
          className="calculatorButton mathButton"
          type="button"
          onClick={() => handleButtonClick('5')}
        >
          5
        </button>
        <button
          className="calculatorButton mathButton"
          type="button"
          onClick={() => handleButtonClick('6')}
        >
          6
        </button>
        <button
          className="mathButton calculatorButton operationButton"
          type="button"
          onClick={() => handleButtonClick('-')}
        >
          -
        </button>
        <button
          className="calculatorButton mathButton"
          type="button"
          onClick={() => handleButtonClick('1')}
        >
          1
        </button>
        <button
          className="calculatorButton mathButton"
          type="button"
          onClick={() => handleButtonClick('2')}
        >
          2
        </button>
        <button
          className="calculatorButton mathButton"
          type="button"
          onClick={() => handleButtonClick('3')}
        >
          3
        </button>
        <button
          className="mathButton calculatorButton operationButton"
          type="button"
          onClick={() => handleButtonClick('+')}
        >
          +
        </button>
        <button
          className="calculatorButton zero mathButton"
          type="button"
          onClick={() => handleButtonClick('0')}
        >
          0
        </button>
        <button
          className="calculatorButton mathButton"
          type="button"
          onClick={() => handleButtonClick('.')}
        >
          .
        </button>
        <button
          className="mathButton calculatorButton equalsButton"
          type="button"
          onClick={() => handleButtonClick('=')}
        >
          =
        </button>
      </div>
    </div>
  );
}

ChildComponent.propTypes = {
  calcData: PropTypes.shape({
    total: PropTypes.string,
    next: PropTypes.string,
    operation: PropTypes.string,
  }).isRequired,
  handleButtonClick: PropTypes.func.isRequired,
};

export default ChildComponent;

```

2. Update the Calculator component this way :point_down: 

```
import React, { useState } from 'react';
import './Calculator.css';
import calculate from '../logic/calculate';
import ChildComponent from './ChildComponent';

const Calculator = () => {
  const [calcData, setCalcData] = useState({
    total: null,
    next: null,
    operation: null,
  });

  const handleButtonClick = (buttonName) => {
    const newData = calculate(calcData, buttonName);
    setCalcData(newData);
  };

  return (
    <div className="container">
      <ChildComponent
        calcData={calcData}
        handleButtonClick={handleButtonClick}
      />
    </div>
  );
};

export default Calculator;

```

### I hope this helps 🤗

**Review by ridwanediallo — CHANGES_REQUESTED** [review]
# Changes Required ♻️
## Hi @MarcoDDM , 


AlmostThereReachGIF

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


## (Highlights) Good Points: :+1:
- You documented your work professionally. ✔
- You used GitHub flow correctly. ✔
- Your commit messages are descriptive. ✔


 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by AmaduKamara — APPROVED** [review]
# Status: Approved 🟢 🟢 🟢

Hi @MarcoDDM,

Good job implementing the changes highlighted by the previous reviewer 👍🏽

Your project is complete!  There is nothing else to say other than... it's time to merge it: ship it:
Congratulations! 🎉 

## Highlights:

✅ You used Gitflow correctly 👍🏽
✅ The Linter checks passed successfully 👍🏽
✅ Used calculator logic correctly 👍🏽
✅ The calculator functions work fine as expected 👍🏽

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

**Pull request merged**